src/python/turicreate/toolkits/sound_classifier/sound_classifier.py

```diff
@@ -237,40 +247,25 @@ def create(dataset, target, feature, max_iterations=10,
     train_metric = mx.metric.Accuracy()
     if validation_data:
         validation_metric = mx.metric.Accuracy()
-    softmax_cross_entropy_loss = mx.gluon.loss.SoftmaxCrossEntropyLoss()
+
     for i in range(max_iterations):
         # TODO: early stopping
 
         for batch in train_data:
-            data = mx.gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0, even_split=False)
-            label = mx.gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0, even_split=False)
-
-            # Inside training scope
-            with mx.autograd.record():
-                for x, y in zip(data, label):
-                    z = custom_NN(x)
-                    # Computes softmax cross entropy loss.
-                    loss = softmax_cross_entropy_loss(z, y)
-                    # Backpropagate the error for one iteration.
-                    loss.backward()
-            # Make one step of parameter update. Trainer needs to know the
-            # batch size of data to normalize the gradient by 1/batch_size.
-            trainer.step(batch.data[0].shape[0])
+            mxnet_model.train(batch)
         train_data.reset()
 
         # Calculate training metric
         for batch in train_data:
-            data = mx.gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0, even_split=False)
-            label = mx.gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0, even_split=False)
-            outputs = [custom_NN(x) for x in data]
+            data, label = mxnet_model.batch_process(batch)
```

Comment:
Let's move this inside `train` / `predict`. We don't want anything specific to MXNet backends here.

Reply:
The reason I left this here was because we don't want to move metric in the mxnet class and `label` is used by `train_metric`